slsim/FalsePositives/false_positive_pop.py

```diff
@@ -0,0 +1,124 @@
+        :param weights: A list of weights corresponding to the probabilities of 
+         selecting each value in source_number_choice. If None, all choices are equally 
+         likely. Defaults to None.
+        """
+
+        self.cosmo = cosmo
+        self._lens_galaxies = elliptical_galaxy_population
+        self._sources = blue_galaxy_population
+        self._choice = source_number_choice
+        self._weights = weights_for_source_number
+        self.los_config = los_config
+        if self.los_config is None:
+            self.los_config = LOSConfig()
+    
+    def draw_deflector(self):
+        """Draw and prepare a deflector (lens) with tolerance-based z_max.
+        
+        :return: a deflector instance and deflector redshift with tolerance added.
+        """
+        lens = self._lens_galaxies.draw_deflector()
+        deflector = Deflector(
+            deflector_type=self._lens_galaxies.deflector_profile,
+            deflector_dict=lens,
+        )
+        z_max = deflector.redshift + 0.002  # Adding tolerance to redshift
+        return deflector, z_max
+
+    def draw_sources(self, z_max):
+        """Draw source(s) within the redshift limit of z_max.
+        :param z_max: maximum redshift for drawn source.
+        :return: A Source instance or a list of Source instance.
+
+        """
+        source_number = random.choices(self._choice, weights=self._weights)[0]
+        source_list = []
+
+        for _ in range(source_number):
+            source = self._sources.draw_source(z_max=z_max)
+            # If no source is available, return None
+            if source is None:
+                return None
+            source_list.append(
+                Source(
+                    source_dict=source,
+                    cosmo=self.cosmo,
+                    source_type=self._sources.source_type,
+                    light_profile=self._sources.light_profile,
+                )
+            )
+        if source_number==1:
+            sources = source_list[0]
+        else:
+            sources = source_list
+        return sources
+
+    def draw_false_positive(self, number=1):
+        
+        """Draw given number of false positives within the cuts of the lens and source.
+
+        :param number: number of false positive requested. The default value is 1.
+        :return: list of FalsePositive() instance.
+         """
+        false_positive_population = []
+
+        for _ in range(number):
+            successful = False
+            while not successful:
+                # Step 1: Draw deflector
+                lens, z_max = self.draw_deflector()
+                # Step 2: Draw sources
+                source = self.draw_sources(z_max)
+                if source is None:
+                    continue  # Retry if sources are invalid
+
+                # Step 3: Create false positive
+                vd=lens.velocity_dispersion(cosmo=self.cosmo)
+                test_area = 3 * draw_test_area(
```

Comment:
is this a bit arbitrary? The area and the interloper space density might be chosen by the user instead.